cmake/linker/ld/target.cmake

```diff
@@ -7,7 +7,9 @@ set(CMAKE_LINKER ${GNULD})
 set_ifndef(LINKERFLAGPREFIX -Wl)
 
 if(NOT "${ZEPHYR_TOOLCHAIN_VARIANT}" STREQUAL "host")
-  if(CONFIG_CPP_EXCEPTIONS)
+  if(CONFIG_CPP_EXCEPTIONS AND
+     (EXISTS "${LIBGCC_DIR}/crtbegin.o") AND
+     (EXISTS "${LIBGCC_DIR}/crtend.o"))
```

Comment:
To me this looks as a very fragile test.

Could we do this another / cleaner way ?

Reply:
I could change it to test if `arch` is `posix` and `CONFIG_LLVM_USE_LD` is enabled. I have only see it for `native_posix` as llvm links `crt{being,end}S.o` automatically. Let me try that.

Reply:
I removed this commit as it is no longer an issue on my machine. The error I had was when I did not have the full list of (other) commits. Having the other commits together probably have fixed the original issue.

Reply:
Changed this to test if `LIBGCC_DIR` is defined. Since posix arch does not have it, it skips including `crtbegin.o` and `crtend.o`.